src/main/java/com/aws/greengrass/clientdevices/auth/session/SessionManager.java

```diff
@@ -40,7 +41,17 @@ protected boolean removeEldestEntry(Map.Entry<String, Session> eldest) {
                 }
             });
 
-    private SessionConfig sessionConfig;
+    private final SessionConfig sessionConfig;
+
+    /**
+     * Construct Session Manager.
+     *
+     * @param sessionConfig Session configuration
+     */
+    @Inject
+    public SessionManager(SessionConfig sessionConfig) {
```

Comment:
I liked how we where setting it before rather than injecting it. The problem with injecting it is that now we can't make this immutable given injected instances are singletons.